sanic/server.py

```diff
@@ -221,12 +221,13 @@ def trigger_events(events, loop):
                 loop.run_until_complete(result)
 
 
-def serve(host, port, request_handler, error_handler, before_start=None,
-          after_start=None, before_stop=None, after_stop=None,
-          debug=False, request_timeout=60, sock=None,
+def serve(protocol, host, port, request_handler, error_handler,
```

Comment:
Would it be smart to put this at the front of the arguments list and possibly break functionality for anyone who uses this? I think putting this at the end and defaulting it to our own HttpProtocol would be better in this case.